react/MuiCozyTheme/theme.jsx

```diff
@@ -465,27 +509,47 @@ const makeOverrides = theme => ({
   MuiDialogTitle: {
     root: {
       ...theme.typography.h3,
-      width: 'calc(100% - (64px + 24px))', // remove padding and close button width + margin
-      padding: '24px 32px',
+      width: dialogConf.desktop.title.width.full,
+      paddingTop: dialogConstants.desktop.paddingTop,
+      paddingBottom: dialogConstants.desktop.paddingBottom,
+      paddingLeft: dialogConf.desktop.title.padding.withoutButton,
+      paddingRight: dialogConf.desktop.title.padding.withoutButton,
       [theme.breakpoints.down('sm')]: {
         ...theme.typography.h4,
-        width: 'calc(100% - 32px)', // remove padding
-        padding: '13px 16px 12px'
+        width: dialogConf.mobile.title.width.full,
+        paddingTop: dialogConstants.mobile.paddingTop,
+        paddingBottom: dialogConstants.mobile.paddingBottom,
+        paddingLeft: dialogConf.mobile.title.padding.withoutButton,
+        paddingRight: dialogConf.mobile.title.padding.withoutButton
       },
-      '&.dialogTitleFluid': {
-        width: '100%',
-        paddingTop: 0,
-        paddingLeft: 0,
-        paddingRight: 0,
+      '&.dialogTitleWithBack': {
+        width: dialogConf.desktop.title.width.withOneButton,
+        paddingLeft: dialogConf.desktop.title.padding.withButton,
         [theme.breakpoints.down('sm')]: {
-          padding: '0 0 24px 0'
+          width: dialogConf.mobile.title.width.withOneButton,
+          paddingLeft: dialogConf.mobile.title.padding.withButton
         }
       },
-      '&.dialogTitleFull': {
-        width: 'calc(100% - 64px)', // remove padding
+      '&.dialogTitleWithClose': {
+        width: dialogConf.desktop.title.width.withOneButton,
+        paddingRight: dialogConf.desktop.title.padding.withButton,
         [theme.breakpoints.down('sm')]: {
-          width: 'calc(100% - 32px)' // remove padding
+          width: dialogConf.mobile.title.width.withOneButton,
+          paddingRight: dialogConf.mobile.title.padding.withButton
         }
+      },
+      '&.dialogTitleWithBackAndClose': {
+        width: dialogConf.desktop.title.width.withTwoButtons,
+        paddingLeft: dialogConf.desktop.title.padding.withButton,
+        paddingRight: dialogConf.desktop.title.padding.withButton,
+        [theme.breakpoints.down('sm')]: {
+          width: dialogConf.mobile.title.width.withTwoButtons,
+          paddingLeft: dialogConf.mobile.title.padding.withButton,
+          paddingRight: dialogConf.mobile.title.padding.withButton
+        }
+      },
+      '&.dialogTitleFluid': {
+        paddingTop: 0
```

Comment:
ça reste compliqué à comprendre de cette manière 🤔 on pourrait peut-être n'avoir qu'un seul point d'entrée au lieu de 2 (dialogConf et dialogConstants). Et peut-être séparer le point d'entrée pour le mobile et desktop ? Donc au final avoir un dialogConfDesktop et dialogConfMobile ?

Reply:
> on pourrait peut-être n'avoir qu'un seul point d'entrée au lieu de 2 (dialogConf et dialogConstants)

Je peux recopier les valeurs de dialogConstants dans dialogConf si tu trouve ça mieux, mais ça n'enlèvera pas le besoin d'avoir 2 objets pour que dialogConf puisse se calculer en fonction des constantes. Ou alors il faudrait jouer avec des méthodes internes à dialogConf et des this, ça se fait mais je trouve ça moins lisible

> Et peut-être séparer le point d'entrée pour le mobile et desktop ? Donc au final avoir un dialogConfDesktop et dialogConfMobile ?

A part supprimer un `.` je ne suis pas sûr de voir ce que ça va apporter?

Reply:
Je réalise que tu as dû faire ton commentaire avant la simplification de box-sizing: https://github.com/cozy/cozy-ui/pull/1905/commits/0357621a1c8773010d19167103be82d378cd8cd3

Car depuis dialogConstants constant n'existe plus, et avec ma mémoire je t'ai répondu comme si c'était encore le cas 🙃

Reply:
Avec les derniers commits ce bloc est devenu suffisament simple pour remettre les anciennes valeurs texte.

As-tu une pref?
```js
const dialogConstants = {
  mobile: {
    paddingHBase: '1rem',
    paddingHWithButton: '3rem', // paddingHBase (1rem) + buttonWidth (1rem) + buttonMargin (1rem)
    paddingV: '0.75rem'
  },
  desktop: {
    paddingHBase: '2rem',
    paddingHWithButton: '4rem', // paddingHBase (2rem) + buttonWidth (1rem) + buttonMargin (1rem)
    paddingV: '1.5rem'
  }
}
//...
MuiDialogTitle: {
    root: {
      ...theme.typography.h3,
      boxSizing: 'border-box',
      width: '100%',
      paddingTop: dialogConstants.desktop.paddingV,
      paddingBottom: dialogConstants.desktop.paddingV,
      paddingLeft: dialogConstants.desktop.paddingHBase,
      paddingRight: dialogConstants.desktop.paddingHBase,
      [theme.breakpoints.down('sm')]: {
        ...theme.typography.h4,
        paddingTop: dialogConstants.mobile.paddingV,
        paddingBottom: dialogConstants.mobile.paddingV,
        paddingLeft: dialogConstants.mobile.paddingHBase,
        paddingRight: dialogConstants.mobile.paddingHBase
      },
      '&.dialogTitleWithBack': {
        paddingLeft: dialogConstants.desktop.paddingHWithButton,
        [theme.breakpoints.down('sm')]: {
          paddingLeft: dialogConstants.mobile.paddingHWithButton
        }
      },
      '&.dialogTitleWithClose': {
        paddingRight: dialogConstants.desktop.paddingHWithButton,
        [theme.breakpoints.down('sm')]: {
          paddingRight: dialogConstants.mobile.paddingHWithButton
        }
      },
      '&.dialogTitleFluid': {
        paddingTop: 0
      }
    }
  },
```

ou
```js

  MuiDialogTitle: {
    root: {
      ...theme.typography.h3,
      boxSizing: 'border-box',
      width: '100%',
      padding: '1.5rem 2rem',
      [theme.breakpoints.down('sm')]: {
        ...theme.typography.h4,
        padding: '0.75rem 1rem',
      },
      '&.dialogTitleWithBack': {
        paddingLeft: '4rem',
        [theme.breakpoints.down('sm')]: {
          paddingLeft: '3rem'
        }
      },
      '&.dialogTitleWithClose': {
        paddingRight: '4rem',
        [theme.breakpoints.down('sm')]: {
          paddingRight: '3rem'
        }
      },
      '&.dialogTitleFluid': {
        paddingTop: 0
      }
    }
  },
```

Reply:
le 2. S'il y a un lien entre les valeurs, l'indiquer en commentaire comme on faisait avant